### PR TITLE
Only allow one worker to start the registry

### PR DESCRIPTION
**Pull request by bunchesofdonald**
The issue this seems to have been running into was that all of the workers were trying to start the registry and all but the first one would be denied by docker itself. This led to the `registry` fixture throwing an exception and the tests relying on the fixture to fail.

This updates the `registry` fixture to only start the container if we're on `gw0` and all other workers will wait for the registry to be available before continuing.

### CODE REVIEW

File: tests/fixtures/docker.py [hunk 1]
```diff
@@ -110,26 +113,55 @@ def prefect_base_image(pytestconfig: "pytest.Config", docker: DockerClient):
...
+
+def pytest_sessionfinish(session, exitstatus):
+    worker_id = os.getenv("PYTEST_XDIST_WORKER", "gw0")
+    if worker_id == "gw0":
+        docker = session.config._docker_client
         try:
-            yield "http://localhost:5555"
-        finally:
+            container = docker.containers.get("orion-test-registry")
             container.remove(force=True)
```
Comment (zzstoatzz) [inline tests/fixtures/docker.py:165]: hmm why do we need this _and_ `remove=True` on L148?

**Review by zzstoatzz — APPROVED** [review]
🏅

File: tests/fixtures/docker.py (hunk 1 above)
Comment (bunchesofdonald) [reply tests/fixtures/docker.py:165]: Oh I should actually remove the `L148` one, thanks!

Comment (bunchesofdonald) [reply tests/fixtures/docker.py:165]: Hmm, actually this isn't working correctly anyway. I'll find a different approach.

**Comment by bunchesofdonald** [comment]
Managing the container lifecycle this way is just too complex, going a different route. Will open a new PR in a bit.

**Pull request closed without merging**